install/updates/40-delegation.update

```diff
@@ -211,6 +211,8 @@ default:ipapermissiontype: SYSTEM
 
 dn: cn=config
 add:aci: (targetattr = "cn || createtimestamp || entryusn || modifytimestamp || objectclass || passsyncmanagersdns*")(target = "ldap:///cn=ipa_pwd_extop,cn=plugins,cn=config")(version 3.0;acl "permission:Read PassSync Managers Configuration";allow (compare,read,search) groupdn = "ldap:///cn=Read PassSync Managers Configuration,cn=permissions,cn=pbac,$SUFFIX";)
+add:aci: (targetattr = "cn || objectclass || nsslapd-changelogmaxentries || nsslapd-changelogmaxage || nsslapd-changelogtrim-interval || nsslapd-encryptionalgorithm || nsSymmetricKey")(targetfilter = "cn=changelog")(target = "ldap:///cn=ldbm database,cn=plugins,cn=config")(version 3.0; acl "Replication Admin read access to replication changelog"; allow (read,search) groupdn = "ldap:///cn=Replication Administrators,cn=privileges,cn=pbac,$SUFFIX";)
+add:aci: (targetattr = "nsslapd-changelogmaxentries || nsslapd-changelogmaxage || nsslapd-changelogtrim-interval || nsslapd-encryptionalgorithm || nsSymmetricKey")(targetfilter = "cn=changelog")(target = "ldap:///cn=ldbm database,cn=plugins,cn=config")(version 3.0; acl "Replication Admin write access to replication changelog"; allow (write) groupdn = "ldap:///cn=Replication Administrators,cn=privileges,cn=pbac,$SUFFIX";)
```

Comment:
The group dn need to be in cn=permissions, not cn=privileges.

The ACI name needs the prefix "permission" ala acl "permission:Replication Admin read access to replication changelog"

Is targetfilter necessary? It isn't really a filter is it? Can't cn=changelog be merged with target?

Reply:
> The group dn need to be in cn=permissions, not cn=privileges.

Well it needs to be one of these groups:

memberOf: cn=Replication Administrators,cn=privileges,cn=pbac,dc=ipa,dc=test
memberOf: cn=Add Replication Agreements,cn=permissions,cn=pbac,dc=ipa,dc=test
memberOf: cn=Modify Replication Agreements,cn=permissions,cn=pbac,dc=ipa,dc=te
 st
memberOf: cn=Read Replication Agreements,cn=permissions,cn=pbac,dc=ipa,dc=test
memberOf: cn=Remove Replication Agreements,cn=permissions,cn=pbac,dc=ipa,dc=te
 st
memberOf: cn=Modify DNA Range,cn=permissions,cn=pbac,dc=ipa,dc=test
memberOf: cn=Read PassSync Managers Configuration,cn=permissions,cn=pbac,dc=ip
 a,dc=test
memberOf: cn=Modify PassSync Managers Configuration,cn=permissions,cn=pbac,dc=
 ipa,dc=test
memberOf: cn=Read LDBM Database Configuration,cn=permissions,cn=pbac,dc=ipa,dc
 =test
memberOf: cn=Add Configuration Sub-Entries,cn=permissions,cn=pbac,dc=ipa,dc=te
 st
memberOf: cn=Read DNA Range,cn=permissions,cn=pbac,dc=ipa,dc=test

Which group would you prefer?

> 
> The ACI name needs the prefix "permission" ala acl "permission:Replication Admin read access to replication changelog"
> 
> Is targetfilter necessary? It isn't really a filter is it? Can't cn=changelog be merged with target?

It's to single out each changelog entry for each backend

cn=changelog,cn=userroot,cn=ldbm database,cn=plugins,cn=config
cn=changelog,cn=ipaca,cn=ldbm database,cn=plugins,cn=config

So we need a filter it so we only target the changelog entry from each backend, and not the backend entry itself:  cn=userroot,cn=ldbm database,cn=plugins,cn=config.  So the ACI targets cn=ldbm database,cn=plugins,cn=config, then it only applies to "cn=changelog" entries under that target.

Reply:
> Which group would you prefer?

I left out this group which is another option:    cn=ipaservers,cn=hostgroups,cn=accounts,$SUFFIX

Reply:
There is a 1-1 mapping of ACI to permission in IPA. A privilege represents an operation and is generally a single permission/ACI but there are times where multiple are required. Examples are adding a user and setting a password are two permissions in the same privilege, and some of the replication management is centralized in a single privilege.

For your purposes, see Read PassSync Managers Configuration in the same update file as an example. The permissions entry is created with a member entry for the Replication Administrators privilege. Then the ACI points to this permission as the bind rule. So you'd need to generate two permissions entries, one for each ACI.